File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& arr) {
        vector<int>ans;
        int count=0;
        for(int i=0;i<arr.size();i++){
            if(arr[i]!=0) ans.push_back(arr[i]);
            if(arr[i]==0) count++;
        }
        for(int i=0;i<count;i++){
            ans.push_back(0);
        }
        arr=ans;

    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] arr) {
        int zero_size=0;
        for(int i =0;i<arr.length;i++){
            if(arr[i]==0)zero_size++;
            else if (zero_size > 0){
                int temp = arr[i];
                arr[i-zero_size] = temp;
                arr[i] = 0;
            }
        }
    }
}